Add docs for PingOne OIDC (#7347)
cloudflare/cloudflare-docs: content/cloudflare-one/identity/idp-integration/pingone-oidc.md

---
pcx_content_type: how-to
title: PingOne®
weight: 7
---

# PingOne®

The PingOne® cloud platform from PingIdentity provides SSO identity management. Cloudflare Access supports PingOne as an OIDC identity provider.

## Set up PingOne as an identity provider

1. In your PingIdentity environment, navigate to **Connections** > **Applications**.

> 2. Click **Add Application**.

Reviewer: ```suggestion
2. Select **Add Application**.
```

3. Enter an **Application Name**.
4. Select **OIDC Web App**.

> 5. Click **Save**.

Reviewer: ```suggestion
5. Select **Save**.
```

> 6. Click **Resource Access** and add the **email** and **profile** scopes.

Reviewer: ```suggestion
6. Select **Resource Access** and add the **email** and **profile** scopes.
```

> 7. In the **Configuration** tab, click **General** and save the Client ID, Client Secret, and Environment IDs. These ids will be used in a later step to add PingOne to the Zero Trust Dashboard.

Reviewer: ```suggestion
7. In the **Configuration** tab, select **General**.
8. Save the **Client ID**, **Client Secret**, and **Environment ID**. These ids will be used in a later step to add PingOne to the Zero Trust Dashboard.
```

> 8. In the **Configuration** tab, click the pencil in the top right corner and enter your [team domain](/cloudflare-one/glossary/#team-domain) in **Redirect URIs** and click **Save**.

Reviewer: ```suggestion
8. In the **Configuration** tab, select the pencil icon.
9. In the **Redirect URIs** field, enter your [team domain](/cloudflare-one/glossary/#team-domain) and select **Save**.
```

> 9. On the [Zero Trust Dashboard](https://dash.teams.cloudflare.com/), navigate to **Settings** > **Authentication**.

Reviewer: ```suggestion
10. On the [Zero Trust Dashboard](https://dash.teams.cloudflare.com/), navigate to **Settings** > **Authentication**.
```

> 10. Under **Login methods**, click **Add new**.

Reviewer: ```suggestion
10. Under **Login methods**, select **Add new**.
```

> 11. Select PingOne.

Reviewer: ```suggestion
11. Select **PingOne**.
```

> 12. Input the Client ID, Client Secret, and Environment ID fields generated previously.

Reviewer: ```suggestion
12. Input the **Client ID**, **Client Secret**, and **Environment ID** generated previously.
```

> 13. (Optional) Enable [Proof of Key Exchange (PKCE)](https://www.oauth.com/oauth2-servers/pkce/). PKCE will be performed on all login attempts.

Reviewer: ```suggestion
14. (Optional) Enable [Proof of Key Exchange (PKCE)](https://www.oauth.com/oauth2-servers/pkce/). PKCE will be performed on all login attempts.
```

> 14. Click **Save**.

Reviewer: ```suggestion
15. Select **Save**.
```
